extensions/auth/portability-auth-google/src/main/java/org/dataportabilityproject/auth/google/GoogleAuthDataGenerator.java

```diff
@@ -44,10 +50,38 @@
  * authorization code and posts it against the AD API to obtain a token for querying the Graph API.
  */
 public class GoogleAuthDataGenerator implements AuthDataGenerator {
-  // TODO: Reduce requested scopes by service and authorization mode (readwrite/read)
-  private static final ImmutableCollection<String> SCOPES =
-      ImmutableSet.of(CalendarScopes.CALENDAR, PeopleServiceScopes.CONTACTS, TasksScopes.TASKS);
+  // The scopes necessary to import or export each supported data type.
+  // Scopes for EXPORT should contain READONLY permissions
+  private static final Map<String, ListMultimap<AuthMode, String>> DATA_TYPE_SCOPES =
```

Comment:
Very nice! Since there's so many levels to this map, can we have two map constants, EXPORT_DATA_SCOPE_TYPES and IMPORT_DATA_SCOPE TYPES?

Reply:
Done